### PR TITLE
refactor(ivy): cleanup of the view compiler

### PR DESCRIPTION
**Pull request by vicb**

### CODE REVIEW

**Comment by vicb** [comment]
OCL:192838588:BASE:192839084:1523660959764:93d61576

**Review by mhevery — APPROVED** [review]

File: packages/compiler/test/render3/r3_compiler_compliance_spec.ts [hunk 1]
```diff
@@ -610,7 +610,7 @@ describe('compiler compliance', () => {
           factory: function ComplexComponent_Factory() { return new ComplexComponent(); },
           template: function ComplexComponent_Template(rf: IDENT, ctx: IDENT) {
             if (rf & 1) {
-              $r3$.ɵpD(0, $c1$);
+              $r3$.ɵpD(0, $c1$, ['span[title=toFirst]', 'span[title=toSecond]']);
```
Comment (mhevery) [inline packages/compiler/test/render3/r3_compiler_compliance_spec.ts]: constant pool.

**Review by mhevery — APPROVED** [review]
but please fix the constant pool thing.

**Comment by vicb** [comment]
http://test/OCL:192838588:BASE:192845806:1523664248804:9f587742

**Pull request closed without merging**